File: Back/AccountService/Models/UserProfile.cs
using System.ComponentModel.DataAnnotations;
using AccountService.Helpers;
using GlobalHelpers;
using GlobalModels;
using MongoDB.Entities;

namespace AccountService.Models;

public class UserProfile : Entity
{
    [Required]
    public Guid UserId { get; private set; }
    public string? City { get; private set; }
    public string Country { get; private set; }
    public List<Contact>? Contacts { get;private set; }
    public string? About { get;private set; }
    public byte[]? Avatar { get;private set; }
    
    public string? FirstName { get;private set; }
    
    public string? LastName { get;private set; }

    public UserProfile(UserProfileToAddDto userProfileToAddDto,(string?, string?) names ,Guid userId)
    {
        UserId = userId;
        City = userProfileToAddDto.City;
        Country = userProfileToAddDto.Country;
        Contacts = userProfileToAddDto.Contacts ?? new List<Contact>();
        About = userProfileToAddDto.About;
        Avatar = userProfileToAddDto.Avatar ?? GetDefaultAvatar();
        FirstName = names.Item1 ?? "";
        LastName = names.Item2 ?? "";
    }
    
    public void UpdateProfile(UserProfileToUpdateDto updatedUserProfile)
    {
        if(!Validation.CheckIfWord(FirstName))
            throw new ArgumentException("Invalid first name");
        
        if(!Validation.CheckIfWord(LastName))
            throw new ArgumentException("Invalid last name");
        
        City = updatedUserProfile.City;
        Country = updatedUserProfile.Country ?? throw new ArgumentNullException(nameof(updatedUserProfile.Country));
        Contacts = updatedUserProfile.Contacts;
        About = updatedUserProfile.About;
        Avatar = updatedUserProfile.Avatar;
        FirstName = updatedUserProfile.FirstName;
        LastName = updatedUserProfile.LastName;
    }
    
    public static byte[] GetDefaultAvatar() => File.ReadAllBytes("Media/UserDefaultAvatar.png");
}

File: Front/lib/Types/UserProfile/UserProfile.ts
import { Contact } from '../Contact';
import IUserProfile from "@/lib/Types/UserProfile/IUserProfile";
export default class UserProfile implements IUserProfile {
    public Id: string;
    Avatar?: Blob | undefined;
    City: string;
    Country: string;
    Contacts: Contact[];
    FirstName?: string;
    LastName?: string;
    About?: string;
    constructor(id: string, city: string, country: string, contact: Contact[], avatar?: Blob | undefined, firstName?: string, lastName?: string, about?: string) {
        if (!id)
            console.log('Id is null');
        
        this.Id = id;
        this.City = city;
        this.Country = country;
        this.Contacts = contact;
        this.Avatar = avatar;
        this.FirstName = firstName;
        this.LastName = lastName;
        this.About = about;
    }
    
    
    public toString(): string { 
        let stringBuilder : string[] = [];
        stringBuilder.push('Id: ' + this.Id);
        stringBuilder.push('City: ' + this.City);
        stringBuilder.push('Country: ' + this.Country);
        stringBuilder.push('Contacts: ' + this.Contacts);
        stringBuilder.push('Avatar: ' + this.Avatar ? 'Avatar is not null' : 'Avatar is null');
        stringBuilder.push('FirstName: ' + this.FirstName);
        stringBuilder.push('LastName: ' + this.LastName);
        stringBuilder.push('About: ' + this.About);
        return stringBuilder.join('\n');
    }
    
    
    
}